### PR TITLE
Initial mode picker

### LINKED ISSUE #15: Support multiple modes of input
_e.g._ shape mode, freeform lines, block select/move

### PR DESCRIPTION
**Pull request by newsch**
I made a rough mode picker interface to use brush mode with, and refactored some of the mode code in the process.

Closes #15.

TODO:
- [x] get feedback
- ~~add mode-level configs~~ see #52
- [x] consider init/teardown methods for modes
- [x] for convenience, #37 should be merged before this
- [x] comment decisions/changes

### CODE REVIEW

**Comment by newsch** [comment]
In action:
[asciicast](https://asciinema.org/a/M3ys1XLI2hAx0A8wGY5I2LNqW)

**Comment by newsch** [comment]
@MatthewBeaudouinLafon and I talked about some next steps: making `mode_master` more separated from "modes", init/teardown options, and reworking the status window.

**Comment by newsch** [comment]
Rebased off of new master

**Comment by newsch** [comment]
I refactored some more, 132004a1ff90c64e38e813699ded79e8243f7960 has the details.

**Comment by newsch** [comment]
Ready for a look-through, those last two are just formatting changes.

File: src/fe_modes.c [hunk 1]
```diff
@@ -105,29 +142,51 @@ int free_line_arrows_to_char(int last_arrow, int current_arrow) {
...
-  // Mode Switch - Enter to canvas,
-  if (state->ch_in == KEY_ENTER) {
-    state->current_mode = state->last_canvas_mode;
+int mode_picker(reason_t reason, State *state) {
+  if (reason == END) {
     return 0;
   }
 
-  // LR Arrows navigation
+  int mode_start = MODE_PICKER + 1;
```
Comment (labseven) [inline src/fe_modes.c]:
```suggestion
// mode_start and num_modes are the bounds of the Mode_ID, used for enumerating all modes. 
  int mode_start = MODE_PICKER + 1;
```

Maybe call it `mode_first`

**Review by labseven — APPROVED** [review]
Looks good. I added a comment

**Comment by newsch** [comment]
Added some changes from comments, this should be good to go

**Pull request merged**